### ISSUE TITLE
Configure fails inside nix-shell --pure

### ISSUE THREAD

**Issue by yangm97** [matrix-construct/construct#57]
Using the following `shell.nix`

```
with import <nixpkgs> {};

stdenv.mkDerivation {
  name = "construct";

  buildInputs = [
    pkgs.git
    pkgs.automake
    pkgs.autoconf
    pkgs.libtool
    pkgs.libsodium
    pkgs.openssl
    pkgs.file
    pkgs.boost
  ];

}
```

**Comment by yangm97** [comment]
```
checking openssl/tls1.h usability... no
checking openssl/tls1.h presence... yes
configure: WARNING: openssl/tls1.h: present but cannot be compiled
configure: WARNING: openssl/tls1.h:     check for missing prerequisite headers?
configure: WARNING: openssl/tls1.h: see the Autoconf documentation
configure: WARNING: openssl/tls1.h:     section "Present But Cannot Be Compiled"
configure: WARNING: openssl/tls1.h: proceeding with the compiler's result
checking for openssl/tls1.h... no
checking for SSL_version in -lssl... yes
checking for OPENSSL_init in -lcrypto... yes
checking magic.h usability... yes
checking magic.h presence... yes
checking for magic.h... yes
checking for magic_version in -lmagic... yes
checking zlib.h usability... no
checking zlib.h presence... no
checking for zlib.h... no
checking for zlibVersion in -lz... no
checking lz4.h usability... no
checking lz4.h presence... no
checking for lz4.h... no
checking for LZ4_versionNumber in -llz4... no
checking snappy.h usability... no
checking snappy.h presence... no
checking for snappy.h... no
checking for snappy_compress in -lsnappy... no
checking for GMP... no
checking whether you asked to use boost sources included here... no
checking for boostlib >= 1.66... yes
checking whether the Boost::ASIO library is available... yes
configure: error: Could not find a version of the library!
```

**Comment by bqv** [comment]
```nix
with import <nixpkgs> {};

stdenv.mkDerivation {
  pname = "matrix-construct";
  version = "unstable";
  src = fetchgit {
    url = "https://github.com/jevolk/charybdis";
    #rev = "e089103a30128986e976ec7264c2162cd9c6ba02";
    sha256 = "1y7a7p7krbi2rnimzjbwr868skmrb129cq0jx42zvakalvbfy48w";
  };
  configureFlags = [
    "--enable-generic"
    "--enable-jemalloc"
    "--with-boost-libdir=${boost.out}/lib"
    "--with-boost=${boost.dev}"
  ];
  nativeBuildInputs = [ autoreconfHook pkg-config ];
  buildInputs = [
    libsodium
    openssl
    file
    graphicsmagick
    boost
    (rocksdb.overrideAttrs (super: rec {
      version = "5.16.6";
      src = fetchFromGitHub {
        owner = "facebook";
        repo = "rocksdb";
        rev = "v${version}";
        sha256 = "0yy09myzbi99qdmh2c2mxlddr12pwxzh66ym1y6raaqglrsmax66";
      };
      NIX_CFLAGS_COMPILE = "${super.NIX_CFLAGS_COMPILE} -Wno-error=redundant-move";
    }))
    zlib
    lz4
    jemalloc
    gmp
    snappy
    llvm
  ];
}
```

**Comment by bqv** [comment]
See jevolk#9

**Comment by yangm97** [comment]
Fixed by 68b47956ef5949623be0d772b1480a398edb99e2
